bindings/experimental/distrdf/python/DistRDF/Proxy.py

```diff
@@ -129,6 +128,13 @@ def _call_action_result(self, *args, **kwargs):
         """
         return getattr(self.GetValue(), self._cur_attr)(*args, **kwargs)
 
+    def create_variations(self):
+        """VariationsFor can only be called on an action node."""
+        newnode = VariationsNode(operation=Operation.create_op("VariationsFor"), get_head=self.proxied_node.get_head)
+        self.proxied_node.children.append(newnode)
```

Comment:
Ok so this means that in the DistRDF graph you will have an action node for `Vary`, for which you add here a child node of type `VariationsFor` (so this is the first case of an action node having a child)?

Reply:
Mmm I just saw `Vary` is classified as transformation, why so?

Reply:
Slightly more subtle than that. `Vary` is a transformation, and in distrdf follows exactly the same pattern as other transformations. `VariationsFor` is what escapes this pattern because it's not a method of `RInterface` like any other RDF operation. So the logic in this function actually means that we are creating a node of the computationgraph for `VariationsFor` (`Vary` will have its own node with a `TransformationProxy` just like `Define` or `Filter`). This special `VariationsNode` will indeed by a child node of the previous node of the distrdf computation graph, which would in turn be proxied by an `ActionProxy`. From the RDF point of view, `VariationsFor` is not even a node, but it creates internally as many nodes as variations when called

Reply:
Ah I see, `Vary` is indeed a transformation. Then why does `VariationsFor` receive an action? Shouldn't it receive a result pointer?